packages/fxa-auth-server/lib/subhub.js

```diff
@@ -0,0 +1,315 @@
+      path: '/customer/:uid/subscriptions',
+      method: 'GET',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+        },
+        // TODO: update with final subscriptions schema from subhub
+        response: isA.array().items(isA.object({
+          plan_id: isA.string().required(),
+          product_id: isA.string().required(),
+          current_period_end: isA.number().required(),
+          end_at: isA.number().required(),
+        }))
+      }
+    },
+
+    getCustomer: {
+      path: '/customer/:uid',
+      method: 'GET',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+        },
+        // TODO: update with final customer schema from subhub
+        response: isA.object()
+      }
+    },
+
+    updateCustomer: {
+      path: '/customer/:uid',
+      method: 'POST',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+        },
+        payload: {
+          pmt_token: isA.string().required(),
+        },
+        response: isA.alternatives(
+          // TODO: update with final customer schema from subhub
+          isA.object(),
+          isA.object({
+            message: isA.string()
+          })
+        )
+      }
+    },
+
+    createSubscription: {
+      path: '/customer/:uid/subscriptions',
+      method: 'POST',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+        },
+        payload: {
+          pmt_token: isA.string().required(),
+          plan_id: isA.string().required(),
+          email: isA.string().required(),
+        },
+        response: isA.alternatives(
+          isA.object({
+            sub_id: isA.string()
+          }),
+          isA.object({
+            message: isA.string()
+          })
+        )
+      }
+    },
+
+    cancelSubscription: {
+      path: '/customer/:uid/subscriptions/:sub_id',
+      method: 'DELETE',
+      validate: {
+        params: {
+          uid: isA.string().required(),
+          sub_id: isA.string().required(),
+        },
+        response: isA.alternatives(
+          isA.object({}),
+          isA.object({
+            message: isA.string()
+          })
+        )
+      }
+    },
+
+  });
+
+  const api = new SubHubAPI(
+    config.subhub.url,
+    {
+      headers: {
+        // TODO: update with subhub final auth
+        Authorization: `Bearer ${config.subhub.key}`
+      },
+      timeout: 15000
+    }
+  );
+
+  return {
+    isStubAPI: false,
+
+    async listPlans() {
+      return api.listPlans();
+    },
+
+    async listSubscriptions(uid) {
+      try {
+        return await api.listSubscriptions(uid);
+      } catch (err) {
+        if (err.statusCode === 404) {
+          log.error('subhub.listSubscriptions.1', { uid, err });
+          // TODO: update with subhub listSubscriptions error response for invalid uid
+          if (err.message === 'invalid uid') {
+            throw error.unknownAccount(uid);
```

Comment:
`unknownAccount` seems a bit strange here because the account is known to *us* otherwise this method wouldn't be called, it's just not known by subhub. Maybe another [new error number is warranted](https://github.com/mozilla/fxa/pull/819/files#diff-fd7dff0f3afc526facd6365fed66bcb7R1039) to indicate "no known subscriptions".

Reply:
Yeah, [this is an open question](https://github.com/mozilla/subhub/issues/3) I have on subhub: Is there a difference between "no known subscriptions" or "never been a customer yet" or "was a customer, cancelled all subscriptions" - but are all of them the same error / response?

Reply:
For now, I created a new `unknownCustomer` error to capture this. It may change, depending on what subhub eventually does (issue #719 )